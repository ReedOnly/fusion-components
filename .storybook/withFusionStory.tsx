import React from 'react';
import Header from '../src/components/core/Header';
import Content from '../src/components/core/Content';
import Overlay from '../src/components/general/ApplicationGuidance/components/Overlay';

export default (title: string, padding: number = 16) => (stories: () => React.ReactNode) => {
    return (
        <>
            <Overlay fixed>
                <Header aside={null} start={null} content={null} />
            </Overlay>
            <Content>
                <Overlay>
                    <div
                        style={{
                            padding,
                            minHeight: '100%',
                            width: '100%',
                            maxWidth: '100%',
                            boxSizing: 'border-box',
                        }}
                    >
                        {stories()}
                    </div>
                </Overlay>
            </Content>
        </>
    );
};
