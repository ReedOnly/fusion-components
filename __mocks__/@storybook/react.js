import renderer from 'react-test-renderer';
import addons, { mockChannel } from '@storybook/addons';
import { withFusionContext } from '../../.storybook/withFusionContext';

addons.setChannel(mockChannel());

export const action = (actionName) => jest.fn();
const fusionContext = withFusionContext();
export const storiesOf = (groupName) => {
    const api = {
        decorators: [],
        parameters: {},

        add(storyName, story) {
            if (api.parameters.skipTest) {
                return;
            }

            describe(groupName, () => {
                it(storyName, () => {
                    // const decoratedStory = api.decorators.reduce((s, decorator) => {
                    //     try {
                    //         return decorator(s, { parameters: api.parameters });
                    //     } catch (e) {
                    //         return s;
                    //     }
                    // }, story);

                    const component = renderer.create(fusionContext(story));

                    // @TODO Enable custom names once released:
                    // > https://github.com/facebook/jest/pull/2094
                    expect(component.toJSON())
                        .toMatchSnapshot
                        // `${groupName}.${storyName}`
                        ();
                });
            });

            return api;
        },

        addDecorator() {
            return api;
        },

        addParameters(parameters) {
            api.parameters = { ...api.parameters, ...parameters };
            return api;
        },
    };

    return api;
};

export const addDecorator = () => {};
