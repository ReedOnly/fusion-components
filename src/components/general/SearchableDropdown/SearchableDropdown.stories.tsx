import * as React from 'react';
import { storiesOf } from '@storybook/react';
import SearchableDropdownWrapper, { SearchableDropdownOption } from './index';
import withFusionStory from '../../../../.storybook/withFusionStory';

const dropdownOptions = [
    {
        key: '1',
        title: 'First',
    },
    {
        key: '2',
        title: 'Selected',
        isSelected: true,
    },
    {
        key: '3',
        title: 'Disabled',
        isDisabled: true,
    },
    {
        key: '4',
        title: 'Last',
    },
];

const DropdownStory = () => {
    const [options, setOptions] = React.useState<SearchableDropdownOption[]>(dropdownOptions);
    const [optionsNoLAbel, setOptionsNoLabel] = React.useState<SearchableDropdownOption[]>(
        dropdownOptions
    );

    const updateOptions = item =>
        options.map(option => {
            return { ...option, isSelected: item.key === option.key };
        });

    return (
        <div style={{ margin: '8px' }}>
            <SearchableDropdownWrapper
                options={options}
                label="Dropdown"
                onSelect={item => setOptions(updateOptions(item))}
            />
            <br />
            <SearchableDropdownWrapper
                options={optionsNoLAbel}
                onSelect={item => setOptionsNoLabel(updateOptions(item))}
            />
        </div>
    );
};

storiesOf('General|Searchable Dropdown', module)
    .addDecorator(withFusionStory('Searchable Dropdown'))
    .add('Default', () => <DropdownStory />);
