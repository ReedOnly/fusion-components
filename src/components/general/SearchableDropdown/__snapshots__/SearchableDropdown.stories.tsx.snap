// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General|Searchable Dropdown Default 1`] = `
<div
  className="container comfortable"
>
  <div
    className="container comfortable"
  >
    <div
      style={
        Object {
          "margin": "8px",
        }
      }
    >
      <div>
        <div
          className="inputContainer"
        >
          <div
            className="inputContent error"
            onClick={[Function]}
          >
            <div
              className="inputTextContent error"
            >
              <label>
                Dropdown
              </label>
              <input
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyUp={[Function]}
                placeholder=""
                value=""
              />
            </div>
            <div
              className="icon"
              onClick={[Function]}
            >
              <svg
                height={24}
                style={
                  Object {
                    "color": "var(--color-red)",
                  }
                }
                viewBox="0 0 24 24"
                width={24}
              >
                <path
                  clipRule="evenodd"
                  d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM12 13C11.45 13 11 12.55 11 12V8C11 7.45 11.45 7 12 7C12.55 7 13 7.45 13 8V12C13 12.55 12.55 13 12 13ZM11 15V17H13V15H11Z"
                  fill="currentColor"
                  fillRule="evenodd"
                />
              </svg>
            </div>
          </div>
          <div
            className="helperText error"
          >
            required
          </div>
        </div>
      </div>
      <br />
      <div>
        <div
          className="inputContainer"
        >
          <div
            className="inputContent labelLess"
            onClick={[Function]}
          >
            <div
              className="inputTextContent"
            >
              <input
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyUp={[Function]}
                placeholder="Type to search..."
                value=""
              />
            </div>
            <div
              className="icon"
              onClick={[Function]}
            >
              <svg
                cursor="pointer"
                height={24}
                style={
                  Object {
                    "color": "",
                  }
                }
                viewBox="0 0 24 24"
                width={24}
              >
                <path
                  d="M8.70627 11.4137L11.2963 14.0037C11.6863 14.3937 12.3163 14.3937 12.7063 14.0037L15.2963 11.4137C15.9263 10.7837 15.4763 9.70374 14.5863 9.70374H9.40627C8.51627 9.70374 8.07627 10.7837 8.70627 11.4137Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <br />
      <div>
        <div
          className="inputContainer"
        >
          <div
            className="inputContent"
            onClick={[Function]}
          >
            <div
              className="inputTextContent"
            >
              <label>
                Select food
              </label>
              <input
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyUp={[Function]}
                placeholder=""
                value=""
              />
            </div>
            <div
              className="icon"
              onClick={[Function]}
            >
              <svg
                cursor="pointer"
                height={24}
                style={
                  Object {
                    "color": "",
                  }
                }
                viewBox="0 0 24 24"
                width={24}
              >
                <path
                  d="M8.70627 11.4137L11.2963 14.0037C11.6863 14.3937 12.3163 14.3937 12.7063 14.0037L15.2963 11.4137C15.9263 10.7837 15.4763 9.70374 14.5863 9.70374H9.40627C8.51627 9.70374 8.07627 10.7837 8.70627 11.4137Z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <br />
      <div
        style={
          Object {
            "width": "300px",
          }
        }
      >
        <div>
          <div
            className="inputContainer"
          >
            <div
              className="inputContent"
              onClick={[Function]}
            >
              <div
                className="inputTextContent"
              >
                <label>
                  Narrow food
                </label>
                <input
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyUp={[Function]}
                  placeholder=""
                  value=""
                />
              </div>
              <div
                className="icon"
                onClick={[Function]}
              >
                <svg
                  cursor="pointer"
                  height={24}
                  style={
                    Object {
                      "color": "",
                    }
                  }
                  viewBox="0 0 24 24"
                  width={24}
                >
                  <path
                    d="M8.70627 11.4137L11.2963 14.0037C11.6863 14.3937 12.3163 14.3937 12.7063 14.0037L15.2963 11.4137C15.9263 10.7837 15.4763 9.70374 14.5863 9.70374H9.40627C8.51627 9.70374 8.07627 10.7837 8.70627 11.4137Z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="overlay"
  />
  <div
    className="snacks"
  />
  <div
    className="dialog"
  />
</div>
`;
